core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java

```diff
@@ -452,6 +452,11 @@ private SignupInfo validateAccountCreationForm(StaplerRequest req, boolean valid
             si.errors.put("password1", Messages.HudsonPrivateSecurityRealm_CreateAccount_PasswordRequired());
         }
 
+        if (FIPS140.useCompliantAlgorithms()) {
+            if (si.password1.length() < 14) {
```

Comment:
I would suggest to move literal magic number 14 to some appropriately named constant and use that constant instead.